cmd/syntactic-code-intel-worker/shared/config.go

```diff
@@ -5,36 +5,43 @@ import (
 	"strconv"
 	"time"
 
-	"github.com/sourcegraph/sourcegraph/internal/codeintel/shared/lsifuploadstore"
 	"github.com/sourcegraph/sourcegraph/internal/env"
 	"github.com/sourcegraph/sourcegraph/lib/errors"
 )
 
+type IndexingWorkerConfig struct {
+	env.BaseConfig
+	PollInterval         time.Duration
+	Concurrency          int
+	MaximumRuntimePerJob time.Duration
+	CliPath              string
+}
+
 type Config struct {
 	env.BaseConfig
 
-	WorkerPollInterval    time.Duration
-	WorkerConcurrency     int
-	WorkerBudget          int64
-	MaximumRuntimePerJob  time.Duration
-	SCIPUploadStoreConfig *lsifuploadstore.Config
-	CliPath               string
-	ListenAddress         string
+	IndexingWorker *IndexingWorkerConfig
```

Comment:
```suggestion
	*IndexingWorkerConfig
```

The usage site reads a bit weird with just `IndexingWorker.Validate()`. It's also a bit confusing because the worker itself is conceptually a different entity from its configuration.